src/sdl2/render.rs

```diff
@@ -790,19 +790,19 @@ impl<'a> Renderer<'a> {
     ///
     /// Errors if drawing fails for any reason (e.g. driver failure),
     /// or if the provided texture does not belong to the renderer.
-    pub fn copy(&mut self, texture: &Texture, src: Option<Rect>, dst: Option<Rect>)
+    pub fn copy<RECT: Into<Option<Rect>>>(&mut self, texture: &Texture, src: RECT, dst: RECT)
```

Comment:
That's a detail but please do not use the same RECT twice here. Doing that makes it a must to have the same type for src and dst, and you might not want this. For instance with your current code you can write:

```rust
renderer.copy(&texture, Some(Rect::new(0,0,10,10)), Some(Rect::new(0,0,10,10)));
renderer.copy(&texture, Rect::new(0,0,10,10), Rect::new(0,0,10,10));
```

but not 

```rust
renderer.copy(&texture, Some(Rect::new(0,0,10,10)), Rect::new(0,0,10,10));
renderer.copy(&texture, Rect::new(0,0,10,10), None);
```

Just make a `Rect1: Into<Option<Rect>>` and `Rect2: Into<Option<Rect>>` and assign Rect1 to src and Rect2 to dst and that'll do the trick !